### PR TITLE
RPCs refactor

### PR DESCRIPTION
**Pull request by mariano54**
This is a major refactor of how the RPCs work.

* All RPC apis must return a Dict, return None, or throw an exception. 
1. If a Dict is returned, the rpc_server automatically adds "success": True to the response, in both HTTP and WebSockets.
2. If None is returned, the rpc_server returns success:True for HTTP, but sends no message for WebSockets
3. If an exception is thrown, the rpc_server returns a good HTTP error code, and success:False, and WebSocket returns "success": False
* wallet_rpc_api.py has been cleaned up. Prefer to throw exceptions rather than returning Optionals everywhere, or returning "success": False. 
* The transaction api calls (send_transaction, cc_spend, and send_clawback_transaction) now return instantly, as opposed to waiting for mempool inclusion. This simplifies the logic significantly. When the full node responds, we send an update via WebSocket (update_tx). For HTTP, I have added the get_transaction API call for this purpose. The client can keep calling and check the "sent_to" field to see if the transaction is included in the mempool.
* All generate_signed_transaction and related APIs in the wallets have been refactored to return a value as opposed to Optional. This is simpler and allows for passing an error code as to why the transaction failed. ie. throw ValueError("Not enough balance"). 
* The RL wallet has been refactored to use types and to properly check for None when it has to.
* Addresses and puzzle hashes are different things. If it's bytes use the term puzzle hash, if it's txch1 use address
* Many breaking RPC changes, mostly in wallet
* Added wallet_rpc_client.py calls for all wallet operations (cc wallet, trades, and rl_wallet are still missing).
* Added tests for wallet_rpc_client which cover the above operations
* Added "chia wallet" program, to replace chia show -w. It only shows balances but can be easily extended to make transactions etc.
* Fix a bug in Block.js that was causing fetching header.js repeatedly
* Fix a bug in WSM that was trying to regenerate Rate Limited puzzles continuously, this was slowing down wallet sync significantly if you had an RL wallet
* Fix a bug with duplicate send_to entries in TransactionRecords
* xch_target_puzzle_hash has been changed to xch_target_address in config
* Moved some mempool related files from util to full node
* Fix farmer screen balance
* Improve performance of wallet sync by not checking for new puzzle hashes

TODO
* Test the rl_wallet and cc_wallet in the UI more extensively. Also test backups

### CODE REVIEW

**Review by hoffmang9 — APPROVED** [review]
With black and a spelling fix that I've merged - looks good.

File: tests/rpc/test_wallet_rpc.py [hunk 1]
```diff
@@ -0,0 +1,190 @@
+import asyncio
```
Comment (Yostra) [inline tests/rpc/test_wallet_rpc.py:1]: 👍👍👍

File: src/wallet/rl_wallet/rl_wallet.py [hunk 2]
```diff
@@ -370,7 +368,15 @@ async def get_pending_change_balance(self) -> uint64:
 
         return uint64(addition_amount)
 
-    def get_new_puzzle(self):
+    def get_new_puzzle(self) -> Program:
+        if (
```
Comment (Yostra) [inline src/wallet/rl_wallet/rl_wallet.py:372]: iirc there's rl_info.initialized that can be used for a single if here

Comment (Yostra) [reply src/wallet/rl_wallet/rl_wallet.py:372]: Oh or this the flake8 thing

File: CODE_OF_CONDUCT.md [hunk 3]
```diff
@@ -24,7 +24,7 @@ Examples of unacceptable behavior by participants include:
 
 * The use of sexualized language or imagery and unwelcome sexual attention or
  advances
-* Excessive trolling, insulting/derogatory comments, and personal or political attacks
+* Insulting/derogatory comments, and personal or political attacks, or excessive trolling.
```
Comment (Yostra) [inline CODE_OF_CONDUCT.md:27]: Tiny amount of trolling is ok ? :)

**Review by Yostra — APPROVED** [review]

File: CODE_OF_CONDUCT.md (hunk 3 above)
Comment (mariano54) [reply CODE_OF_CONDUCT.md:27]: Yes just a bit is funny and beneficial

**Pull request merged**